student/src/main/java/tds/student/services/ReviewTestServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+import TDS.Shared.Exceptions.ReturnStatusException;
+import com.google.common.base.Optional;
+import org.apache.http.HttpStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+import tds.common.ValidationError;
+import tds.student.services.abstractions.IOpportunityService;
+import tds.student.services.data.PageList;
+import tds.student.services.data.TestOpportunity;
+import tds.student.sql.data.OpportunityStatusChange;
+import tds.student.sql.data.OpportunityStatusType;
+import tds.student.sql.repository.remote.ExamRepository;
+import tds.student.web.TestManager;
+
+@Service
+@Scope("prototype")
+public class ReviewTestServiceImpl implements ReviewTestService {
+    private final IOpportunityService opportunityService;
+    private final ExamRepository examRepository;
+    private final ITDSLogger tdsLogger;
+    private final boolean isLegacyCallsEnabled;
+    private final boolean isRemoteCallsEnabled;
+
+    @Autowired
+    public ReviewTestServiceImpl(@Qualifier("integrationOpportunityService") final IOpportunityService opportunityService,
+                                 final ExamRepository examRepository,
+                                 final ITDSLogger tdsLogger,
+                                 @Value("${tds.exam.legacy.enabled}") final boolean legacyCallsEnabled,
+                                 @Value("${tds.exam.remote.enabled}") final boolean remoteExamCallsEnabled) {
+        if (!remoteExamCallsEnabled && !legacyCallsEnabled) {
+            throw new IllegalStateException("Remote and legacy calls are both disabled.  Please check progman configuration for 'tds.exam.remote.enabled' and 'tds.exam.legacy.enabled' settings");
+        }
+
+        this.opportunityService = opportunityService;
+        this.examRepository = examRepository;
+        this.tdsLogger = tdsLogger;
+        isLegacyCallsEnabled = legacyCallsEnabled;
+        isRemoteCallsEnabled = remoteExamCallsEnabled;
+    }
+
+    @Override
+    public ResponseData<String> reviewTest(final TestOpportunity testOpportunity,
+                                           final HttpServletRequest request) throws ReturnStatusException, IOException {
+        ResponseData<String> responseData = new ResponseData<>(TDSReplyCode.OK.getCode(), "OK", null);
+
+        if (isLegacyCallsEnabled) {
+            responseData = legacyReviewTest(testOpportunity, request);
+        }
+
+        if (!isRemoteCallsEnabled) {
+            return responseData;
+        }
+
+        final Optional<ValidationError> maybeValidationError = examRepository.reviewExam(testOpportunity.getOppInstance());
+        if (maybeValidationError.isPresent()) {
```

Comment:
To me this block belongs in the Web layer not the service layer.  An error should be sent back from this service and then the controller can do with it what it wants.  In this example it would send the error.

For example, what if we wanted to use this service deep in another service that doesn't have anything to do with HTTP?